### PR TITLE
Adding ethtool tests for Avocado [v5]

### PR DESCRIPTION
**Pull request by narasimhan-v**
Signed-off-by: Narasimhan V sim@linux.vnet.ibm.com

### CODE REVIEW

File: io/net/ethtool/ethtool_tests.sh [hunk 1]
```diff
@@ -30,7 +30,6 @@ if SoftwareManager().check_installed("ethtool") is False:
         print "Not able to install ethtool"
         exit(1)
 END
-exit
```
Comment (apahim) [inline io/net/ethtool/ethtool_tests.sh]: Could you please squash this commit into the other one?

File: io/net/ethtool/ethtool_tests.sh [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+echo "Parameter: ${arg}"
+
+ifup $interface
+for command in ifdown ifup; do
+    sleep 10
+    $command $interface
+    sleep 10
+    echo "Ethool, $command:"
+    output_file="$AVOCADO_TEST_OUTPUTDIR/output_$command"
+    ethtool ${arg} ${interface} $action_elapse | tee "$output_file"
```
Comment (apahim) [inline io/net/ethtool/ethtool_tests.sh:51]: This is not redirecting the stderr to the `$output_file`. In consequence, the the `elif` in line 57 will not find the _Operation not supported_ message (the same applies to the other `elif`s).

File: io/net/ethtool/ethtool_tests.sh [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    error=${PIPESTATUS[0]}
+
+    if [ $error == 0 ]; then
+        avocado_debug "PASS"
+    elif grep "Operation not supported $output_file"; then
```
Comment (apahim) [inline io/net/ethtool/ethtool_tests.sh:56]: This is doomed to hang. You're executing: 

```
$ grep "Operation not supported $output_file"
```

So you are grepping a string nowhere. I mean, the `$output_file` is part of the string you're grepping. It should be:

```
$ grep "Operation not supported" $output_file
```

Sorry not noticing this early, but only now I could get to this point.

Comment (apahim) [inline io/net/ethtool/ethtool_tests.sh:56]: Same applies to the other `grep`s

**Comment by apahim** [comment]
@narasimhan-v did you actually test the `ethtool` non-zero exit conditions? I mean, did you face them all to check if they're working? The **Operation not supported**, the **no stats available**, and so on. Because the way it is now, I cannot even complete the tests. Please verify your script before send the next version.

**Comment by narasimhan-v** [comment]
@apahim sorry for this, have made all necessary changes, tested and created a new PR #98

**Pull request closed without merging**